src/book/03-guides/04-general/04-documentation.mdx

```diff
@@ -0,0 +1,86 @@
+
+When writing modules, the `.hpp` should include a documenting comment for each class, struct, function, method, global variable, member variable and anything extra that you think may need one.
+
+### Directives
+
+These are some common directives:
+
+| Directive                         | Description                                                                                                                                                                                           |
+| --------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `@brief`                          | A brief description of the following code. Every documenting comment should have this. If the first line does not have a directive it is treated as `@brief`.                                         |
+| `@details`                        | All the details that someone using this thing might need to know beyond what is provided in other directives. Comment lines after the initial one but before any directive are treated as `@details`. |
+| `@param[direction] <formal name>` | Details a parameter that is passed to a function. Direction is either `in` or `out` and is usually omitted.                                                                                           |
+| `@return`                         | Details what a function returns.                                                                                                                                                                      |
+| `@retval <value>`                 | Details specific return value meanings, usually for enums, booleans or error codes.                                                                                                                   |
+| `@throws <type name>`             | Details what a function throws.                                                                                                                                                                       |
+| `@tparam <formal name>`           | Details a template parameter, usually a [named requirement](https://en.cppreference.com/w/cpp/named_req) or other requirements that must be fulfilled.                                                |
+| `@author`                         | The author and maintainer of this code. Git blame is unreliable as some changes could just be formatting.                                                                                             |
+
+This is a list of less common directives that may be used, see the [Doxygen](https://www.doxygen.nl/manual/commands.html#cmd_intro) documentation for the full list.
```

Comment:
```suggestion
This is a list of less common directives that may be used, see the [Doxygen](https://www.doxygen.nl/manual/commands.html#cmd_intro) documentation for the full list:
```
Although with the second clause it feels strange to do. But otherwise it feels attached to the table above and doesn't match the 'These are some common directives:' sentence.